src/digodoc_lib/cmt.ml

```diff
@@ -64,12 +64,191 @@ let pp_val_desc ident pp vdesc =
         done;
         Format.fprintf pp "%s" (Buffer.contents buff)
 
-let getVals {cmi_sign; _} = 
-    let vals = List.filter (function | Sig_value _ -> true | _ -> false) cmi_sign in
-    List.map
-        (fun sign ->
-            match sign with
-            | Sig_value (ident, vald, _) -> 
-                infix (Ident.name ident), Format.asprintf "%a" (pp_val_desc ident) vald
-            | _ -> failwith "should not occur")
-        vals
\ No newline at end of file
+let getVals {cmi_sign; _} =
+    (*
+       For documentation on the types matched here see:
+        https://docs.ocaml.pro/docs/LIBRARY.ocamlcommon@ocaml-base-compiler.4.10.0/Types/index.html#type-signature_item
+      *)
+    List.filter_map (function
+            | Sig_value (ident, vald, _) ->
+                Some (infix (Ident.name ident), Format.asprintf "%a" (pp_val_desc ident) vald)
+            | _ -> None)
+        cmi_sign
+
+(** [standardise_output s] Removes and replaces some char occurences *)
+let standardise_output s =
```

Comment:
I believe changing this code to something like :

```ocaml
let standardise_output fmt s =
  String.iter (function
  | ... -> Format.pp_print_char fmt ...
  ) s
```

Would probably make the rest of the code simpler and more efficient. (You would probably change the name of the function along the way)